src/colorizer/Collection.ts

```diff
@@ -224,16 +222,22 @@ export default class Collection {
       throw new Error(`Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': Fetch failed.`);
     }
 
-    const json = await response.json();
-    if (!Array.isArray(json)) {
-      throw new Error(
-        `Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': JSON is not an array.`
-      );
+    let collection: CollectionFile;
+    try {
+      const json = await response.json();
+      collection = updateCollectionVersion(json);
+    } catch (e) {
+      throw new Error(`Could not parse collections JSON data from url '${absoluteCollectionUrl}': '${e}'`);
     }
 
     // Convert JSON array into map
+    if (collection.datasets.length === 0) {
+      throw new Error(
+        `Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': No datasets found.`
```

Comment:
do your tests cover this path?

Reply:
Good catch, they don't! Adding that now.